src/content/docs/courseflow/backend.md

```diff
@@ -0,0 +1,123 @@
+---
+title: Backend API's
+---
+
+### Backend requirements of CourseFlow within OnTrack:
+
+#### Course:
+
+The routes involved with this will be relatively straightforward.
+
+| Endpoint               | Description                                       | Request Body   | Response Body|
+| ---------------------- | --------------------------------------------------| -------------- | -------------|
+| GET /course            | Get all course data                               | None           | Course[]     |
+| GET /course/courseId   | Get a course by id                                | None           | Course       |
+| GET /course/search     | Get courses that partially match the search params| None           | Course[]     |
+| POST /course           | Add a new course.                                 | Course         | Course       |
+| PUT /course/courseId   | Updates an existing course via its id             | Course         | Course       |
+| DELETE /course/courseId| Deletes an existing course via its id             | None           | None         |
+
```

Comment:
Some of these endpoints were modified for clarity and to prevent overlap, please check the PR for courseAPI to see how it was changed